scraper/src/github-scraper/discussion.ts

```diff
@@ -37,54 +36,111 @@ query($org: String!, $cursor: String) {
                   }
                 }
                 createdAt
+                updatedAt
               }
             }
           }
         }
       }
     }
   }
-}
-`;
+}`;
 
-async function fetchDiscussionsForOrg(org: string, cursor = null) {
-  const variables = { org, cursor };
-  const response = await octokit.graphql.paginate(query, variables);
+async function fetchGitHubDiscussions(
+  org: string,
+  endDate: Date,
+  startDate: Date,
+  cursor = null,
+) {
+  const variables = {
+    org,
+    cursor,
+  };
+  for await (const response of octokit.graphql.paginate.iterator(
+    query,
+    variables,
+  )) {
+    const repositories = await response.organization.repositories.edges;
+    type repo = (typeof repositories)[0];
+    for (const repo of repositories) {
+      const discussions = await repo.node.discussions.edges.map(
+        (discussion: repo) => ({
+          repoName: repo.node.name,
+          discussion: discussion.node,
+        }),
+      );
```

Comment:
1. Repeating for the fourth time again... Use proper naming for types.
2. Also, why would you infer type of repo and use it for mapping discussions?
3. Are you sure repoName attribute is correct? 

```suggestion
    for (const repo of repositories) {
      const discussions = await repo.node.discussions.edges.map(
        (repo) => ({
          repoName: repo.node.name,
          discussion: repo.node,
        }),
      );
```